tobius/imagecolors
# Refactor codebase to ES6

## Refactor codebase to ES6

> Refactor codebase to ES6 syntax standards (this codebase is over 2 years old)

### Changes

- Update `.editorconfig` to latest preferences
- Update `eslint` config (support Airbnb and Mocha standards)
- Ignore `node_modules` (eslint) and `*.swp` (gitignore)
- Update `README` language to remind users of ImageMagick requirement
- Update codebase to use ES6 syntax standards
- Fix all newly identified `eslint` errors (to support Airbnb standards)
- Fix unreported bug (discovered during cleanup) caused by histograms with decimal values
- Remove `lodash` dependency (no longer needed due to built-in ES6 capabilities)
- Add `depcheck` devDependency for internal maintenance
- Add `eslint*` devDependencies (for Airbnb configuation)
- Add `npm run dev` to run example code in hot swap mode (for local dev)
- Add `nodemon` devDependency (for `npm run dev` command)
- Update node dependency and devDependency versions (minors and revisions only)

### Semantics

- [ ] This PR will introduce breaking changes (major)
- [ ] This PR will introduce new functionality (minor)
- [x] This PR will fix or tweak existing functionality (patch)

### Categories

- [ ] Configs
- [x] Bugfixes
- [ ] File Structure
- [ ] Provisioning

### Validation Steps

1. Run `npm run test` to ensure that all unit tests pass
2. Run `npm run lint` to ensure that all lint tests pass
3. Run `npm run dev` to ensure that all example pages work

### Deployment Steps

1. Merge
